tests/integration_test.py

```diff
@@ -101,6 +103,13 @@ def tearDown(self):
                 pass
         for folder in self.tmp_folders:
             shutil.rmtree(folder)
+
+        for volume in self.tmp_volumes:
+            try:
+                self.client.remove_volume(volume)
+            except docker.errors.APIError:
+                pass
```

Comment:
This is one place where pytest can really improve the test code. Instead of having every test attempt this, you can use a https://pytest.org/latest/yieldfixture.html#yieldfixture (or https://pytest.org/latest/fixture.html#finalization) in just the tests that use a volume to do the proper cleanup.

We don't have to make that change now, but I think we should move in that direction eventually.